packages/reactive_ble_mobile/ios/Classes/Plugin/PluginController.swift

```diff
@@ -491,6 +497,66 @@ final class PluginController {
         completion(.success(result))
     }
 
+    func writeDescriptorWithoutResponse(name: String, args: WriteDescriptorRequest, completion: @escaping PlatformMethodCompletionHandler) {
+        guard let central = central
+        else {
+            completion(.failure(PluginError.notInitialized.asFlutterError))
+            return
+        }
+        
+        guard let descriptor = QualifiedDescriptorIDFactory().make(from: args.descriptor)
+        else {
+            completion(.failure(PluginError.invalidMethodCall(method: name, details: "descriptor, characteristic, service, and peripheral IDs are required").asFlutterError))
+            return
+        }
+        
+        guard let peripheral = central.activePeripherals[descriptor.peripheralID],
+              let service = peripheral.services?.first(where: { service in
+                  service == descriptor.serviceID
+              }),
+              let characteristic = service.characteristics?.first(where: { characteristic in
+                  characteristic == descriptor.characteristicID
+              }) else {
+            return
+        }
+        
+        if let _ = characteristic.descriptors?.first(where: { $0.uuid == descriptor.id }) {
+            let result = performWriteDescriptorWithoutResponse(central: central, args: args, descriptor: descriptor)
+            completion(.success(result))
+        } else {
+            do {
+                try central.discoverDescriptors(
```

Comment:
@werediver @remonh87 would like your opinion if this is the correct approach. I think there is no need to discover all characteristics from the get go. It may also imply a API change to declare what to discover. 

The code currently discovers on-demand but may change the library behavior from what the use expects?

Reply:
Characteristic discovery works differently on iOS and Android. On iOS a client can specify a list of services and characteristics to discover when connecting to a device (the client is only allowed to interact with the discovered characteristics later on), while on Android no discovery is triggered beforehand (I think it's happening on the fly then, but entirely sure; CC @remonh87).

In terms of design, a simple solution is to extend the existing discovery mechanism to also discover descriptors, as specified when connecting. That would be consistent and quite efficient.

Discovering descriptors one by one is definitely not as efficient (the more descriptors a client uses, the larger the overhead is). This is also not consistent with how the library works with characteristics (on iOS).

Reply:
@werediver go you regarding the differences - and will think how to extend the initial discovery. It does mean the existing iOS discovery code will get another layer (currently counts how many services are left to discover, this will be extended with the number of characteristics to explore).